stellargraph/core/graph.py

```diff
@@ -196,41 +196,53 @@ def has_node(self, node: Any) -> bool:
         """
         raise NotImplementedError
 
-    def neighbors(self, node: Any) -> Iterable[Any]:
+    def neighbors(self, node: Any, weight=False, edge_types=None) -> Iterable[Any]:
         """
         Obtains the collection of neighbouring nodes connected
         to the given node.
 
         Args:
             node (any): The node in question.
+            weight (bool, default False): If True, each neighbour in the output is a
```

Comment:
I think 

>  it could be a filter (i.e. neighbors(weight=10) will find neighbours with weight 10)
is a very different function than what I would expect the `neighbors`, `in_nodes`, `out_nodes` should do.  Given a node it should return all neighbors of the node. Given a node and `weight=True`, it should return all neighbors nodes and the weights on their edges. If `edge_type` is none then, it should return all neighbors of the node with a dictionary of `edge_types` and (list of) weight(s) corresponding the the neighbor edge of the specific `edge_type`. 
And lastly, if the `edge_types` is provided as a list, then the weights for edges for only the types specified.

The filtering of neighborhood based on thresholding the weights could be something seperate. This would be quite handy for the temporal random walks.
However, I think the basic   `neighbors`, `in_nodes`, `out_nodes` should be about returning all neighbors, all in_nodes, all out_nodes with or without weights and for all edge types by default and subset of edge types if indicated.

Reply:
I did consider returning an output format that keeps things organised in terms of `edge_type`, but I didn't think it was necessary for now so I stuck to something simpler